packages/iov-socket/src/queueingstreamingsocket.ts

```diff
@@ -0,0 +1,32 @@
+// tslint:disable:readonly-array readonly-keyword
+import { StreamingSocket } from "./streamingsocket";
+
+/**
+ * A StreamingSocket that can queue requests.
+ */
+export class QueueingStreamingSocket extends StreamingSocket {
```

Comment:
I like the extension design in this case

Reply:
Hmm, on the other hand: this extension allows you to bypass the queue, which is something the API should disallow. So I tend to prefer composition over extension here.